pallets/thea-message-handler/src/lib.rs

```diff
@@ -197,8 +204,8 @@ pub mod pallet {
 			if current_set_id.saturating_add(1) == payload.validator_set_id {
 				<ValidatorSetId<T>>::put(current_set_id.saturating_add(1));
 			}
-
 			<IncomingNonce<T>>::put(payload.nonce);
+			<IncomingMessages<T>>::insert(payload.nonce, payload);
```

Comment:
Do we need store separately nonce in such situation when we are storing it here as a key as well?